bagitobjecttransfer/recordtransfer/forms/transfer_forms.py

```diff
@@ -659,57 +662,35 @@ def clean(self):
 
 class GroupTransferForm(TransferForm):
     def __init__(self, *args, **kwargs):
-        users_groups = kwargs.pop('users_groups')
+        self.user = kwargs.pop('user', None)
         super().__init__(*args, **kwargs)
-        self.fields['group_name'].choices = [
-            ('No Group', gettext('-- None Selected --')),
-            ('Add New Group', gettext('-- Add New Group --')),
-            *[(x.name, x.name) for x in users_groups],
-        ]
-        self.allowed_group_names = [x[0] for x in self.fields['group_name'].choices]
-        self.fields['group_name'].initial = 'No Group'
+        users_groups = SubmissionGroup.objects.filter(created_by=self.user)
+        choices = [(None, 'Select a group')] + [(group.uuid, group.name) for group in users_groups]
+        self.fields['group_id'].choices = choices
+        self.fields['group_id'].initial = None
 
     def clean(self):
         cleaned_data = super().clean()
-        group_name = cleaned_data['group_name']
-        if group_name not in self.allowed_group_names:
-            self.add_error('group_name', f'Group name "{group_name}" was not in list')
-        if group_name == 'Add New Group' and not cleaned_data['new_group_name']:
-            self.add_error('new_group_name', 'Group name cannot be empty')
-        if group_name == 'Add New Group' and not cleaned_data['group_description']:
-            self.add_error('group_description', 'Group description cannot be empty')
+        group_id = cleaned_data.get('group_id')
+
+        users_groups = SubmissionGroup.objects.filter(created_by=self.user)
+        valid_group_ids = users_groups.values_list('uuid', flat=True)
+        if group_id and group_id not in valid_group_ids:
+            self.add_error('group_id', 'Invalid group selected.')
         return cleaned_data
```

Comment:
I suggest simplifying this logic to:

```suggestion
        if group_id and not SubmissionGroup.objects.filter(created_by=self.user, uuid=group_id).exists():
            self.add_error('group_id', 'Invalid group selected.')
        return cleaned_data
```

Reply:
Much cleaner, I like it.

Resolved by: https://github.com/NationalCentreTruthReconciliation/Secure-Record-Transfer/pull/226/commits/2078c7cb0ee06ff3cd8373e6d6a6ae63c587c235